lib/AST/ModuleNameLookup.cpp

```diff
@@ -278,6 +280,39 @@ void ModuleNameLookup<LookupStrategy>::lookupInModule(
   if (decls.size() - initialCount <= 1)
     return;
 
+  // Some functions like `memchr` are defined both in libc and in libc++.
+  // Importing both would result in ambiguities, but there are some attributes
+  // that mark the preferred overloads. See _LIBCPP_PREFERRED_OVERLOAD.
+  llvm::SmallPtrSet<ValueDecl *, 4> declsToRemove;
+  bool hasPreferredOverload = false;
+  for (auto decl : decls)
+    if (const auto *clangDecl = decl->getClangDecl()) {
+      if (clangDecl->hasAttr<clang::EnableIfAttr>()) {
+        bool thisDeclHasPreferredOverload = false;
+        for (auto clangAttr :
+             clangDecl->specific_attrs<clang::EnableIfAttr>()) {
+          if (auto litExpr =
+                  dyn_cast<clang::CXXBoolLiteralExpr>(clangAttr->getCond())) {
+            if (litExpr->getValue()) {
+              thisDeclHasPreferredOverload = hasPreferredOverload = true;
```

Comment:
this keeps checking `enable_if` until one is `true`. The semantics of the attribute is the opposite: the function is only available if *all* of them evaluate to `true`.